lib/nrf_modem_lib/nrf_modem_lib_trace_sync.c

```diff
@@ -6,76 +6,20 @@
 
 #include <stdio.h>
 #include <zephyr/kernel.h>
-#include <zephyr/kernel.h>
-#include <zephyr/drivers/pinctrl.h>
 #include <modem/nrf_modem_lib_trace.h>
+#include <modem/trace_medium.h>
 #include <nrf_modem_at.h>
-#ifdef CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_UART
-#include <nrfx_uarte.h>
-#endif
-#ifdef CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_RTT
-#include <SEGGER_RTT.h>
-#endif
-
-#ifdef CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_UART
-#define UART1_NL DT_NODELABEL(uart1)
-PINCTRL_DT_DEFINE(UART1_NL);
-static const nrfx_uarte_t uarte_inst = NRFX_UARTE_INSTANCE(1);
-#endif
-
-#ifdef CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_RTT
-static int trace_rtt_channel;
-static char rtt_buffer[CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_RTT_BUF_SIZE];
-#endif
+
 
 static bool is_transport_initialized;
 static bool is_stopped;
 
-#ifdef CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_UART
-static bool uart_init(void)
-{
-	int ret;
-	const nrfx_uarte_config_t config = {
-		.skip_gpio_cfg = true,
-		.skip_psel_cfg = true,
-
-		.hal_cfg.hwfc = NRF_UARTE_HWFC_DISABLED,
-		.hal_cfg.parity = NRF_UARTE_PARITY_EXCLUDED,
-		.baudrate = NRF_UARTE_BAUDRATE_1000000,
-
-		.interrupt_priority = NRFX_UARTE_DEFAULT_CONFIG_IRQ_PRIORITY,
-		.p_context = NULL,
-	};
-
-	ret = pinctrl_apply_state(PINCTRL_DT_DEV_CONFIG_GET(UART1_NL), PINCTRL_STATE_DEFAULT);
-	__ASSERT_NO_MSG(ret == 0);
-
-	return (nrfx_uarte_init(&uarte_inst, &config, NULL) == NRFX_SUCCESS);
-}
-#endif
-
-#ifdef CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_RTT
-static bool rtt_init(void)
-{
-	trace_rtt_channel = SEGGER_RTT_AllocUpBuffer("modem_trace", rtt_buffer, sizeof(rtt_buffer),
-						     SEGGER_RTT_MODE_NO_BLOCK_SKIP);
-
-	return (trace_rtt_channel > 0);
-}
-#endif
-
 int nrf_modem_lib_trace_init(struct k_heap *trace_heap)
 {
 	ARG_UNUSED(trace_heap);
 	is_stopped = false;
 
-#ifdef CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_UART
-	is_transport_initialized = uart_init();
-#endif
-#ifdef CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_RTT
-	is_transport_initialized = rtt_init();
-#endif
-
+	is_transport_initialized = trace_medium_init();
```

Comment:
Should we align the `trace_medium_(de)init` with NCS retvals?
```suggestion
	is_transport_initialized = trace_medium_init() == 0 ? true : false;
```

Reply:
Yes

Reply:
fixed